src/packagedcode/phpcomposer.py

```diff
@@ -135,23 +146,30 @@ def is_phpcomposer_json(location):
     return (filetype.is_file(location) and fileutils.file_name(location).lower() == 'composer.json')
 
 
+def is_phpcomposer_lock(location):
+    return (filetype.is_file(location) and fileutils.file_name(location).lower() == 'composer.lock')
+
+
 def parse(location):
     """
     Return a Package object from a composer.json file or None.
     Note that this is NOT exactly the packagist .json format and this is NOT the
     composer.lock format (allare closely related of course but have important
     (even if minor) differences.
     """
-    if not is_phpcomposer_json(location):
-        return
-
-    with io.open(location, encoding='utf-8') as loc:
-        package_data = json.load(loc, object_pairs_hook=OrderedDict)
-
-    return build_package(package_data)
-
-
-def build_package(package_data):
+    is_json = is_phpcomposer_json(location)
```

Comment:
This code below (e.g. the if json or is lock test) likely means that you would be better off with a different class or subclass... But since we are stuck with one class per Package type, I guess that the way to to go for now .
Now this is minor and feel free to ignore this but may be this would be a tad simpler code?
```
if is_phpcomposer_json(location):
    with io.open(location, encoding='utf-8') as loc:
        package_data = json.load(loc, object_pairs_hook=OrderedDict)
    yield build_package_from_json(package_data)

elif is_phpcomposer_lock(location):
    with io.open(location, encoding='utf-8') as loc:
        package_data = json.load(loc, object_pairs_hook=OrderedDict)

    for package in build_package_from_lock(package_data):
        yield package

```